sqsconsumerconfig.go

```diff
@@ -51,14 +45,57 @@ func (c *DefaultSQSQueueConsumerComponent) New(ctx context.Context, config *Defa
 		Endpoint:   aws.String(config.AWSEndpoint),
 	})
 
-	logger, err := c.Logger.New(ctx, config.Logger)
-	if err != nil {
-		return DefaultSQSQueueConsumer{}, err
-	}
-
 	return DefaultSQSQueueConsumer{
-		Logger:   logger,
 		QueueURL: config.QueueURL,
 		Queue:    q,
 	}, nil
 }
+
+// SmartSQSQueueConsumerConfig represents the configuration to configure SmartSQSQueueConsumer
+type SmartSQSQueueConsumerConfig struct {
+	AWSEndpoint     string
+	QueueURL        string
+	QueueRegion     string
+	NumWorkers      uint64
+	MessagePoolSize uint64
+}
+
+// Name of the configuration
+func (*SmartSQSQueueConsumerConfig) Name() string {
+	return "sqsworker"
+}
+
+// SmartSQSQueueConsumerComponent enables creating configured Component
+type SmartSQSQueueConsumerComponent struct {
+	Logger          *log.Component
+	NumWorkers      uint64
+	MessagePoolSize uint64
+}
+
+// NewSmartSQSQueueConsumerComponent generates a new SmartSQSQueueConsumerComponent
+func NewSmartSQSQueueConsumerComponent() *SmartSQSQueueConsumerComponent {
+	return &SmartSQSQueueConsumerComponent{}
+}
+
+// Settings generates the default configuration for DefaultSQSQueueConsumerComponent
+func (c *SmartSQSQueueConsumerComponent) Settings() *SmartSQSQueueConsumerConfig {
+	return &SmartSQSQueueConsumerConfig{}
+}
+
+// New creates a configured SmartSQSConsumer
+func (c *SmartSQSQueueConsumerComponent) New(ctx context.Context, config *SmartSQSQueueConsumerConfig) (SmartSQSConsumer, error) {
+	var sesh = session.Must(session.NewSession())
+	q := sqs.New(sesh, &aws.Config{
+		Region:     aws.String(config.QueueRegion),
+		HTTPClient: http.DefaultClient,
+		Endpoint:   aws.String(config.AWSEndpoint),
+	})
+
+	return SmartSQSConsumer{
+		LogFn:           LoggerFromContext,
```

Comment:
question: doesn't this need to be `log.LoggerFromContext`? Where is this top-level name coming from?

Reply:
`LoggerFromContext` is an alias to `log.LoggerFromContext`